apollo-router/src/services/supergraph_service.rs

```diff
@@ -153,7 +154,26 @@ where
                 QueryPlannerContent::Plan { query, plan } => {
                     let can_be_deferred = plan.root.contains_defer();
 
-                    if let Some(err) = query.validate_variables(body, &schema).err() {
+                    if can_be_deferred && !req.originating_request.headers().get_all(ACCEPT).iter().filter_map(|value| value.to_str().ok()).flat_map(|value| value.split(';')
```

Comment:
Er, sorry, I think I must have confused you a bit here. Comma separates different mime types.  Semicolon is used within a mime type to add parameters to it. So this is actually incorrect now.

like

`accept: text/html; charset=utf-8; text/plain; charset=ascii`

I would suggest going back to comma-splitting and using the same `mime::Mime` parser you use in csrf to parse each comma-separated section.  (This will still be slightly incorrect because technically quoted strings in parameters can have commas, but it will get you most of the way there.) This will be needed anyway because we probably will use a parameter on multipart/mixed to specify that we're implementing the current pre-merged draft spec.

BTW I filed https://github.com/picoHz/mediatype/issues/4 to hopefully make such a parser a thing that exists soon :)

Reply:
yup, rereading the rfc, the , should be used, but the format is not well explained

Reply:
that'll work better with the mediatype crate c613ca8504379b182d55343e8dca807c4ab30817